packages/auth/src/app.ts

```diff
@@ -189,83 +189,117 @@ export class App {
-    }
+    /* Back-channel GNAP Routes */
+    // Grant Initiation
+    this.publicRouter.post(
+      '/',
+      createValidatorMiddleware(openApi, {
+        path: '/',
+        method: HttpMethod.POST
+      }),
+      signatureService.tokenHttpsigMiddleware,
+      grantRoutes.create
+    )
 
-    for (const path in openApi.paths) {
-      for (const method in openApi.paths[path]) {
-        if (isHttpMethod(method)) {
-          let route: (ctx: AppContext) => Promise<void>
-          if (path.includes('continue')) {
-            route = grantRoutes[grantMethodToRoute[method]]
-          } else if (path.includes('token')) {
-            route = accessTokenRoutes[tokenMethodToRoute[method]]
-          } else if (path.includes('introspect')) {
-            this.publicRouter[method](
-              toRouterPath(path),
-              createValidatorMiddleware<ContextType<typeof route>>(openApi, {
-                path,
-                method
-              }),
-              this.config.introspectionHttpsig
-                ? signatureService.tokenHttpsigMiddleware
-                : (ctx, next) => next(),
-              accessTokenRoutes.introspect
-            )
-            continue
-          } else if (path.includes('interact')) {
-            if (path.endsWith('/finish')) {
-              route = grantRoutes.interaction.finish
-            } else {
-              route = grantRoutes.interaction.start
-            }
-          } else if (path.includes('grant')) {
-            if (path.endsWith('/accept')) {
-              route = grantRoutes.interaction.accept
-            } else if (path.endsWith('/reject')) {
-              route = grantRoutes.interaction.reject
-            } else {
-              route = grantRoutes.interaction.details
-            }
-          } else {
-            if (path === '/' && method === HttpMethod.POST) {
-              route = grantRoutes.create
-            } else {
-              this.logger.warn({ path, method }, 'unexpected path/method')
-              continue
-            }
-          }
-          if (route) {
-            this.publicRouter[method](
-              toRouterPath(path),
-              createValidatorMiddleware<ContextType<typeof route>>(openApi, {
-                path,
-                method
-              }),
-              signatureService.tokenHttpsigMiddleware,
-              route
-            )
-            // TODO: remove once all endpoints are implemented
-          } else {
-            this.publicRouter[method](
-              toRouterPath(path),
-              (ctx: AppContext): void => {
-                ctx.status = 200
-              }
-            )
-          }
-        }
-      }
-    }
+    // Grant Continue
+    this.publicRouter.post(
+      '/continue/id',
+      createValidatorMiddleware(openApi, {
+        path: '/continue/{id}',
+        method: HttpMethod.POST
+      }),
+      signatureService.tokenHttpsigMiddleware,
+      grantRoutes.continue
+    )
```

Comment:
The configured (not latest) spec also has `PATCH` and `DELETE` for `/continue/{id}`.
https://github.com/interledger/open-payments/blob/53236cc92a070ec98a7ab31c6ed0f6fc50a98041/auth-server-open-api-spec.yaml#L200-L299
I'm surprised our old magic `for` loop(s) didn't complain :grimacing:

Reply:
Interesting...will update to use the latest spec.